lib/dpl/provider/gae.rb

```diff
@@ -10,6 +10,10 @@ class GAE < Provider
       BOOTSTRAP="#{INSTALL}/#{NAME}/bin/bootstrapping/install.py"
       GCLOUD="#{INSTALL}/#{NAME}/bin/gcloud"
 
+      def with_python_2_7(cmd)
+        context.shell("bash -c 'source #{context.env['HOME']}/virtualenv/python2.7/bin/activate; #{cmd}'")
```

Comment:
1. Since you're invoking bash, I don't think it's necessary to expand `~` yourself anymore.
2. I wonder if you should escape `cmd`. There's no cases now where it's invoked with an apostrophe, but in the future...